### ISSUE TITLE
Detect structure member shadowing another variable

### ISSUE THREAD

**Issue by XVilka** [rizinorg/rizin#1004]
Currently, Rizin can detect if structure overshadows some variables, e.g.:
```
 |           ; var int64_t var_60h @ rsp+0x60
 |           ; var int64_t var_68h @ rsp+0x68
 |           ; var Range<usize> iter @ rsp+0x70
-|           ; var int64_t var_78h @ rsp+0x78
 |           ; var int64_t var_80h @ rsp+0x80
 |           ; var int64_t var_88h @ rsp+0x88
 |           ; var &mut [i32] values @ rsp+0x90
-|           ; var int64_t var_98h @ rsp+0x98
 |           ; var usize val @ rsp+0xa0
 |           ; var usize __next @ rsp+0xa8
 |           ; var usize i @ rsp+0xb0
 |           ; arg int64_t arg1 @ rdi
 |           ; arg int64_t arg2 @ rsi
 |           0x00005270      4881ecb80000.  sub   rsp, 0xb8             ; void bubble_sort<i32>(&mut [i32] values);
 |           0x00005277      4889bc249000.  mov   qword [values.data_ptr], rdi ; arg1
 |           0x0000527f      4889b4249800.  mov   qword [rsp + 0x98], rsi ; arg2
 |           0x00005287      48897c2448     mov   qword [var_48h], rdi  ; arg1
 |           0x0000528c      4889742440     mov   qword [var_40h], rsi  ; arg2
 ```
 Note the  `var int64_t var_98h @ rsp+0x98` was removed in the ` shadow_var_struct_members(RzAnalysisVar *var)` function in `librz/analysis/var.c`.
 But the structure type `_mut__i32_`(`&mut [i32]`) has two members:
 ```
 [0x00005270]> tk~mut__i32
_mut__i32_=struct
struct._mut__i32_=data_ptr,length
struct._mut__i32_.data_ptr=i32 *,0,0
struct._mut__i32_.length=usize,8,0
 ```
 which is equivalent to:
 ```c
 struct _mut__i32_ {
     uint64_t data_ptr;
     uint64_t length;
 };
 ```
 Thus, after removing the `var_98h` it should mark that `mov   qword [rsp + 0x98], rsi` as `mov   qword [values.length], rsi` instead.
 
 There is a test in `test/db/cmd/dwarf` called "function info integration 2 rust". 
 The binary in question is https://github.com/rizinorg/rizin-testbins/blob/master/elf/dwarf_rust_bubble
 
 Related: https://github.com/rizinorg/rizin/pull/1001#issuecomment-817717297

**Comment by thestr4ng3r** [comment]
What is important to note here is that the current/old code only "works" when the type is a struct at the root, so it only works when there is a struct A, it only works when the type of the variable is exactly `A` and only for structs.
So it makes the most sense to fix this after #371 is done, which means that then variables don't just contain a string in C syntax that would have to be re-interpreted every time it is read, but a full `RzType`.

In particular, for direct struct types, this makes it more reliable to determine which struct there is, but it also directly enables the same thing to be done for array types.
For example for a variable `uint64_t myvar[0x10]`, we can then show things like `mov qword [myvar[0x4]], rsi` or for something like `struct A myvar[0x10]` even things like `mov qword [myvar[0x4].somemember], rsi`

**Comment by XVilka** [comment]
Types are updated but there are some problems extracting argument type for `dbg.bubble_sort()` function now:

<img width="1374" alt="Screenshot 2023-06-27 at 14 02 21" src="https://github.com/rizinorg/rizin/assets/203261/95a15b05-d24b-4ee3-b528-a4d69a96f852">
